src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToStringTransformer.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Transforms between a date string and a DateInterval object.
+ *
+ * @author Steffen Roßkamp <steffen.rosskamp@gimmickmedia.de>
+ */
+class DateIntervalToStringTransformer implements DataTransformerInterface
+{
+    /**
+     * Format used for generating strings.
+     *
+     * @var string
+     */
+    private $format;
+
+    /**
+     * Whether to parse by as a signed interval.
+     *
+     * @var bool
+     */
+    private $parseSigned;
+
+    /**
+     * Transforms a \DateInterval instance to a string.
+     *
+     * @see \DateInterval::format() for supported formats
+     *
+     * @param string $format      The date format
+     * @param bool   $parseSigned Whether to parse as a signed interval
+     */
+    public function __construct($format = 'P%yY%mM%dDT%hH%iM%sS', $parseSigned = false)
+    {
+        $this->format = $format;
+        $this->parseSigned = $parseSigned;
+    }
+
+    /**
+     * Transforms a DateInterval object into a date string with the configured format
+     * and timezone.
+     *
+     * @param \DateInterval $value A DateInterval object
+     *
+     * @return string An ISO 8601 or relative date string like date interval presentation
+     *
+     * @throws TransformationFailedException If the given value is not a \DateInterval instance.
+     */
+    public function transform($value)
+    {
+        if (null === $value) {
+            return '';
+        }
+        if (!$value instanceof \DateInterval) {
+            throw new TransformationFailedException('Expected a \DateInterval.');
+        }
+
+        return $value->format($this->format);
+    }
+
+    /**
+     * Transforms a date string in the configured into a DateInterval object.
+     *
+     * @param string $value An ISO 8601 or date string like date interval presentation
+     *
+     * @return \DateInterval An instance of \DateInterval
+     *
+     * @throws TransformationFailedException If the given value is not a string or
+     *                                       if the date interval could not be parsed.
+     */
+    public function reverseTransform($value)
+    {
+        if (empty($value)) {
+            return;
+        }
+        if (!is_string($value)) {
+            throw new TransformationFailedException('Expected a string.');
+        }
+        try {
+            if ($this->isISO8601($value)) {
```

Comment:
@stof 
Just saw again that this throw > catch > throw pattern is also used in DateTimeToStringTransformer. Most probably I got it originally from there.

So maybe both should be changed (DateTimeToStringTransformer in another PR then) or just be left as is.

What do you think?